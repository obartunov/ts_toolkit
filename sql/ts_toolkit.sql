-- Adjust this setting to control where the objects get created.
SET search_path = public;

--
-- Turn off echoing so that expected file
-- does not depend on contents of this file.
--
SET client_min_messages = warning;
\set ECHO none
\i ts_toolkit.sql
\set ECHO all
RESET client_min_messages;

-- ts_score

SELECT ts_score(to_tsvector('english', '1 2 3 4 5'), to_tsquery('english', '1'));
SELECT ts_score(to_tsvector('english', '1 2 3 4 5'), to_tsquery('english', '1&2'));
SELECT ts_score(to_tsvector('english', '1 2 3 4 5'), to_tsquery('english', '1|2'));

SELECT ts_score(to_tsvector('english', '1 2 3 2 1'), to_tsquery('english', '1&2'));
SELECT ts_score(to_tsvector('english', '1 2 3 2 1'), to_tsquery('english', '1|2'));

