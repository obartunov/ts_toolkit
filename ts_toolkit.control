comment = 'ts_score ranking function for fts (combination of ts_rank and ts_rank_cd)'
default_version = '1.0'
module_pathname = '$libdir/ts_toolkit'
relocatable = true
