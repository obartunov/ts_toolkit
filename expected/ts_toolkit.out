-- Adjust this setting to control where the objects get created.
SET search_path = public;
--
-- Turn off echoing so that expected file
-- does not depend on contents of this file.
--
SET client_min_messages = warning;
\set ECHO none
RESET client_min_messages;
-- ts_score
SELECT ts_score(to_tsvector('english', '1 2 3 4 5'), to_tsquery('english', '1'));
 ts_score  
-----------
 0.0607927
(1 row)

SELECT ts_score(to_tsvector('english', '1 2 3 4 5'), to_tsquery('english', '1&2'));
 ts_score 
----------
 0.121585
(1 row)

SELECT ts_score(to_tsvector('english', '1 2 3 4 5'), to_tsquery('english', '1|2'));
 ts_score 
----------
 0.121585
(1 row)

SELECT ts_score(to_tsvector('english', '1 2 3 2 1'), to_tsquery('english', '1&2'));
 ts_score 
----------
 0.243171
(1 row)

SELECT ts_score(to_tsvector('english', '1 2 3 2 1'), to_tsquery('english', '1|2'));
 ts_score 
----------
 0.151982
(1 row)

